src/lib/provable/merkle-list.ts

```diff
@@ -331,7 +331,11 @@ class MerkleListIterator<T> implements MerkleListIteratorBase<T> {
   currentHash: Field;
   currentIndex: Unconstrained<number>;
 
-  constructor(value: MerkleListIteratorBase<T>) {
+  constructor(
+    value: MerkleListIteratorBase<T> & {
+      emptyHash: Field;
+    }
+  ) {
```

Comment:
this doesn't make sense to me, because the empty hash is supposed to be a static property of the Iterator, and not an instance property